components/locale/src/fallback/algorithms.rs

```diff
@@ -149,6 +147,66 @@ impl<'a> LocaleFallbackIteratorInner<'a> {
         locale.region = None;
     }
 
+    fn step_script(&mut self, locale: &mut DataLocale) {
+        // Remove the subdivision keyword
+        if let Some(value) = locale.subdivision.take() {
+            self.backup_subdivision = Some(value);
+            return;
+        }
+        // Remove variants
+        if let Some(variant) = locale.variant.take() {
+            self.backup_variant = Some(variant);
+            return;
+        }
+        // Check for parent override
+        if let Some((language, script, region)) = self.get_explicit_parent(locale) {
+            locale.language = language;
+            locale.script = script;
+            locale.region = region;
+            self.restore_subdivision_variants(locale);
+            return;
+        }
```

Comment:
Issue: I don't think we want the parent overrides in this form. We probably do want parent overrides, but not the language ones.

I think we want:

- `und-Hans` --> `und-Hani`
- `und-Hant` --> `und-Hani`
- _maybe_ `sr-Latn` --> `sr`
- maybe some others, like `bs-Cyrl` -> `bs`

Reply:
We want `nb` -> `no` if we want to use this for collator.

Reply:
ok, I think it's harmless to use the language parents, though we also want script-specific parents.